### PR TITLE
Wb1.1.5.0

### PR DESCRIPTION
**Pull request by AMatutat**
*HUD*
- hud can now be toggled between pixel and grid system with `HUD.usePixelSystem(boolean)` (maybe you have to rescale your HUD-Elements for this).

*DungeonCamera*
- dungeon camera do not focus anymore on `(0,0,0)` if no followable object is set. This allows free camera movements. 
- add `DungeonCamera.setFocusPoint(Point)`. Stops following and set the camera on a fix position

*Bug fixes* 
- fixed issue where firstFrame was called twice if you resize the window in `setup`. This would call `setup` again and loads the first level twice.

### CODE REVIEW

File: core/src/de/fhbielefeld/pmdungeon/vorgaben/graphic/HUD.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
     private final SpriteBatch hudBatch;
     private final OrthographicCamera hudCamera;
     private final List<IHUDElement> hudElements;
+    private boolean usePixelSystem=false;
```
Comment (cagix) [inline core/src/de/fhbielefeld/pmdungeon/vorgaben/graphic/HUD.java:22]: please run some formatter 😎 sometimes you use `type = var;`, some other times you use `type=var;`

**Review by cagix — CHANGES_REQUESTED** [review]
kannst du bitte in der ide nochmal das projekt formatieren?

File: core/src/de/fhbielefeld/pmdungeon/vorgaben/graphic/HUD.java (hunk 1 above)
Comment (AMatutat) [reply core/src/de/fhbielefeld/pmdungeon/vorgaben/graphic/HUD.java:22]: This is for SQ-Project purposes ^^
i should stop doing these things at night.

**Comment by AMatutat** [comment]
Hmm, didn't expect ij to format the `readme.md`, have to take a look on that.

File: README.md [hunk 2]
```diff
@@ -76,37 +100,43 @@ public class Hero implements IAnimatable {
...
-    //Laden der Texturen für die Animation (Pfad angeben)
-    idle.add(new Texture(PATH_TO_TEXTURE_1.png));
-    idle.add(new Texture(PATH_TO_TEXTURE_1.png));
-    //Erstellen einer Animation, als Parameter wird die Liste mit den Texturen
-    //und die Wartezeit (in Frames) zwischen den Wechsel der Texturen angegeben
-    idleAnimation = new Animation(idle,8);
-}
+public Hero(){
+        //Erstellen einer ArrayList
+        List<Texture> idle=new ArrayList<>();
```
Comment (AMatutat) [inline README.md:111]: so here we do `var=value` but in the real code not..... interessting behavior

File: README.md [hunk 3]
```diff
@@ -1,18 +1,18 @@
 ---
 title: "Dokumentation zum PM-Dungeon"
 author: "André Matutat"
-lang: de-DE
-...
-
-
+lang: de-DE ...
```
Comment (AMatutat) [inline README.md:4]: @cagix is this okay?

File: README.md [hunk 4]
```diff
@@ -76,37 +100,43 @@ public class Hero implements IAnimatable {
...
+public Hero(){
+        //Erstellen einer ArrayList
+        List<Texture> idle=new ArrayList<>();
+        //Laden der Texturen für die Animation (Pfad angeben)
+        idle.add(new Texture(PATH_TO_TEXTURE_1.png));
+        idle.add(new Texture(PATH_TO_TEXTURE_1.png));
+        //Erstellen einer Animation, als Parameter wird die Liste mit den Texturen
+        //und die Wartezeit (in Frames) zwischen den Wechsel der Texturen angegeben
+        idleAnimation=new Animation(idle,8);
+        }
```
Comment (AMatutat) [inline README.md:118]: just no.... dont do that

File: README.md [hunk 5]
```diff
@@ -76,37 +100,43 @@ public class Hero implements IAnimatable {
...
 
 @Override
 //Da unser Held aktuell nur eine Animation hat,
 //geben wir diese als aktuell aktive Animation zurück
-public Animation getActiveAnimation() {
-    return this.idle;
-}
+public Animation getActiveAnimation(){
+        return this.idle;
+        }
```
Comment (AMatutat) [inline README.md:125]: arghhhh

File: README.md [hunk 6]
```diff
@@ -134,116 +167,131 @@ private Point position;
...
 Zum Überprüfen, ob ein neues Level geladen werden soll, verwenden wir diesmal die `endFrame`-Methode.
 
 ```java
 @Override
-public void endFrame() {
-    //Prüfe ob der übergebene Point auf der Leiter ist
-    if (levelController.checkForTrigger(hero.getPosition())) {
+public void endFrame(){
+        //Prüfe ob der übergebene Point auf der Leiter ist
+        if(levelController.checkForTrigger(hero.getPosition())){
```
Comment (AMatutat) [inline README.md:237]: again, in real code its `if (blablala)` with a space

File: README.md [hunk 7]
```diff
@@ -134,116 +167,131 @@ private Point position;
...
+Um eine Grafik auf dem HUD anzeigen zu können, erstellen wir zuerst eine neue Klasse, welche das Interface `IHUDElement`
+implementiert.
 
 ```java
 public class HeartIcon implements IHUDElement {
     @Override
     public Point getPosition() {
         //festlegen der Position
-        return new Point(0.5f,4.5f);
+        return new Point(0.5f, 4.5f);
```
Comment (AMatutat) [inline README.md:294]: but why?

File: README.md [hunk 8]
```diff
@@ -273,60 +326,73 @@ public class YourClass extends MainController {
...
-Es werden die Formate `.mp3`, `.wav` und `.ogg` unterstützt. Das Vorgehen unterscheidet sich zwischen den Formaten nicht.
+Es werden die Formate `.mp3`, `.wav` und `.ogg` unterstützt. Das Vorgehen unterscheidet sich zwischen den Formaten
+nicht.
 
 ```java
 //Datei als Sound-Objekt einladen
-Sound bumSound = Gdx.audio.newSound(Gdx.files.internal("assets/sound/bum.mp3"));
-bumSound.play();
+Sound bumSound=Gdx.audio.newSound(Gdx.files.internal("assets/sound/bum.mp3"));
+        bumSound.play();
```
Comment (AMatutat) [inline README.md:357]: guess we have some space left?

File: README.md [hunk 9]
```diff
@@ -273,60 +326,73 @@ public class YourClass extends MainController {
...
+oder ihn vom Bildschirm zu entfernen.
 
 Im unteren Beispiel wird ein Text implementiert, welcher das aktuelle Level ausgibt.
 
 ```java
 public class MyMain extends MainController {
     .....
     Label levelLabel;
-    int levelCounter=0;
+    int levelCounter = 0;
```
Comment (AMatutat) [inline README.md:388]: okay........ guess we take this random sh***

**Comment by cagix** [comment]
wtf, ij is really gready 🤮

File: README.md (hunk 2 above)
Comment (cagix) [reply README.md:111]: nice 😂

File: README.md (hunk 3 above)
Comment (cagix) [reply README.md:4]: this is just a marker for pandoc

File: README.md (hunk 9 above)
Comment (AMatutat) [reply README.md:388]: > yeah, just revert the changes to readme.md (git checkout -- Readme.md)...

can i realy do that AFTER I pushed? 

> is there a switch to exclude this file when formatting the project?

i hope so

**Comment by cagix** [comment]
> after push

well, you *could* revert the change locally and force push the new head to github …

**Comment by cagix** [comment]
> undo ij changes (I know this way is dirty but its also fast) 

if you squash-merge non of this will be visible anymore 😎

**Comment by AMatutat** [comment]
This should be the last major patch for some time.

**Pull request merged**